components/nimbus/src/lib.rs

```diff
@@ -43,3 +53,85 @@ pub use evaluator::TargetingAttributes;
 
 #[cfg(test)]
 mod tests;
+
+impl UniffiCustomTypeConverter for JsonObject {
+    type Builtin = String;
+
+    fn into_custom(val: Self::Builtin) -> uniffi::Result<Self> {
+        let json: Value = serde_json::from_str(&val)?;
+
+        match json.as_object() {
+            Some(obj) => Ok(obj.clone()),
+            _ => Err(anyhow::anyhow!(
+                "Unexpected JSON-non-object in the bagging area"
+            )),
+        }
+    }
+
+    fn from_custom(obj: Self) -> Self::Builtin {
+        Value::Object(obj).to_string()
+    }
+}
+
+cfg_if::cfg_if! {
+    if #[cfg(feature = "stateful-uniffi-bindings")] {
+        include!(concat!(env!("OUT_DIR"), "/nimbus.uniffi.rs"));
+    } else {
+        pub use enrollment::{
+            EnrolledFeatureConfig, ExperimentEnrollment, EnrolledReason, DisqualifiedReason,
+            NotEnrolledReason
+        };
+        use std::collections::HashSet;
+        use std::str::FromStr;
+        use uuid::Uuid;
+
+        impl UniffiCustomTypeConverter for Value {
+            type Builtin = String;
+
+            fn into_custom(val: Self::Builtin) -> uniffi::Result<Self> {
+                let json: Value = serde_json::from_str(&val)?;
+                Ok(json)
+            }
+
+            fn from_custom(val: Self) -> Self::Builtin {
+                val.to_string()
+            }
+        }
+
+        impl UniffiCustomTypeConverter for Uuid {
+            type Builtin = String;
+
+            fn into_custom(val: Self::Builtin) -> uniffi::Result<Self> {
+                let uuid: Uuid = Uuid::from_str(&val)?;
+                Ok(uuid)
+            }
+
+            fn from_custom(val: Self) -> Self::Builtin {
+                val.to_string()
+            }
+        }
+
+        pub type StringHashSet = HashSet<String>;
+
+        impl UniffiCustomTypeConverter for StringHashSet {
+            type Builtin = String;
+
+            fn into_custom(val: Self::Builtin) -> uniffi::Result<Self> {
+                let set: Value = serde_json::from_str(&val)?;
+
+                match set.as_array() {
+                    Some(obj) => Ok(HashSet::from_iter(obj.clone().iter().map(|x| x.as_str().unwrap().to_string()))),
+                    _ => Err(anyhow::anyhow!(
+                        "Unexpected JSON-non-array in the bagging area"
+                    )),
+                }
+            }
+
+            fn from_custom(val: Self) -> Self::Builtin {
+                Value::Array(Vec::from_iter(val).iter().map(|x| Value::String(x.to_owned())).collect()).to_string()
+            }
+        }
```

Comment:
I don't think these converters are necessary once you cut down on the API in the UDL.